app/assets/javascripts/preferences/panes/listed/BlogItem.tsx

```diff
@@ -0,0 +1,79 @@
+import { Button } from '@/components/Button';
+import { HorizontalSeparator } from '@/components/shared/HorizontalSeparator';
+import { LinkButton, Subtitle } from '@/preferences/components';
+import { WebApplication } from '@/ui_models/application';
+import { Action, SNComponent, SNItem } from '@standardnotes/snjs/dist/@types';
+import { FunctionalComponent } from 'preact';
+import { useState } from 'preact/hooks';
+
+type Props = {
+  item: SNComponent;
+  showSeparator: boolean;
+  disabled: boolean;
+  disconnect: (item: SNItem) => void;
+  application: WebApplication;
+};
+
+export const BlogItem: FunctionalComponent<Props> = ({
+  item,
+  showSeparator,
+  disabled,
+  disconnect,
+  application,
+}) => {
+  const applicationAlertService = application.alertService;
+
+  const [isDisconnecting, setIsDisconnecting] = useState(false);
+
+  const handleDisconnect = () => {
+    setIsDisconnecting(true);
+    applicationAlertService
+      .confirm(
+        'Disconnecting will result in loss of access to your blog. Ensure your Listed author key is backed up before uninstalling.',
+        `Disconnect blog "${item.name}"?`,
+        'Disconnect',
+        1
+      )
+      .then((shouldDisconnect) => {
+        if (shouldDisconnect) {
+          disconnect(item);
+        } else {
+          setIsDisconnecting(false);
+        }
+      })
+      .catch((err) => console.error(err));
+  };
+
+  return (
+    <>
+      <Subtitle>{item.name}</Subtitle>
+      <div className="flex">
+        <LinkButton
+          className="mr-2"
+          label="Open Blog"
+          link={
+            (item as any).package_info.actions.find(
```

Comment:
Is it possible here to use a specific type here instead of `any`?

Reply:
`item as SNComponent` most likely (or maybe SNExtension)

Reply:
I've tried both `SNComponent` and `SNActionsExtension` but both give errors: `Property 'actions' does not exist on type 'ComponentPackageInfo'.`

The reason I'm using `package_info.actions` instead of accessing `actions` is because when I'm getting the items using the `getItems()` function, `actions` is empty and gives an error, whereas the actions can be found in `package_info.actions`. This is when the site is running.

Am I doing something wrong here?

Reply:
`item.actions` is empty whereas `item.package_info.actions` has all the correct actions.

Reply:
I tried using `application.actionsManager.getExtensions()` instead of `application.getItems()` as well, but still the `actions` array seems to be empty while `package_info.actions` has the correct actions.

Reply:
Hmm actually I just realized that actions are downloaded for a blog only when you open the Actions menu in the editor pane. So if you install a new Listed blog but never open the Actions menu, then item.package_info.actions will probably be empty as well.

In that case we'll want to do what the Actions menu is doing, which is downloading the actions for an extension, then displaying the Show Blog option.